src/Analysis/Ast/Impl/Analyzer/Symbols/ClassEvaluator.cs

```diff
@@ -54,7 +54,6 @@ internal sealed class ClassEvaluator : MemberEvaluator {
 
                 var bases = ProcessBases();
                 _class.SetBases(bases);
-                _class.DecideGeneric();
```

Comment:
`SetBases` always requires calling `DecideGeneric` so that's why it makes sense to call it there so someone calling `SetBases` elsewhere won't forget. There other case is `SetGenericParameters` which is somewhat different case.

Reply:
```
classType.SetBases(specificBases.Concat(newBases));
classType._parameters = classType._genericParameters.Values.Distinct().OfType<IGenericTypeParameter>().ToList();
classType.DecideGeneric();
```
can become:
```
classType._parameters = classType._genericParameters.Values.Distinct().OfType<IGenericTypeParameter>().ToList();
classType.SetBases(specificBases.Concat(newBases));
```

Reply:
`classType._parameters` is now `SetGenericParameters`. It is generally bad idea to access private members although it is technically possible in the context of the class itself.